src/deepsparse/server/system_logging.py

```diff
@@ -12,24 +12,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from os import getpid
 from typing import Any
 
+import psutil
 from deepsparse import Pipeline
+from deepsparse.loggers import BaseLogger, MetricCategories
 
 
 __all__ = ["log_resource_utilization", "log_request_details"]
 
+RESOURCE_UTILIZATION_IDENTIFIER_PREFIX = "resource_utilization/"
 
-def log_resource_utilization(pipeline: Pipeline, **kwargs: Any):
+
+def log_resource_utilization(
+    server_logger: BaseLogger,
+    prefix: str = RESOURCE_UTILIZATION_IDENTIFIER_PREFIX,
+    **kwargs: Any,
+):
     """
-    Scope for 1.4:
-    - CPU utilization overall
-    - Memory available overall
-    - Memory used overall (shall we continuously log this?
-      this will be a constant value in time)
-    - Number of core used by the pipeline
+    Logs the resource utilization of the server process.
+    This includes:
+    - CPU utilization
+    - Memory utilization
+    - Total memory available
+
+    + any additional kwargs passed in
+    (where key is the identifier and value is the value to log)
+
+    :param server_logger: the logger to log the metrics to
     """
-    pass
+    process = psutil.Process(getpid())
+
+    # A float representing the current system-wide CPU utilization as a percentage
+    cpu_percent = process.cpu_percent()
+    # A float representing process memory utilization as a percentage
+    memory_percent = process.memory_percent()
+    # Total physical memory
+    total_memory_bytes = psutil.virtual_memory().total
+    total_memory_megabytes = total_memory_bytes / 1024 / 1024
+
+    identifier_to_value = {
+        "cpu_utilization_[%]": cpu_percent,
+        "memory_utilization_[%]": memory_percent,
+        "total_memory_available_[MB]": total_memory_megabytes,
```

Comment:
let's avoid using symbols as words and also reserve brackets in identifiers for slicing
```suggestion
        "cpu_utilization_percent": cpu_percent,
        "memory_utilization_percent": memory_percent,
        "total_memory_available_MB": total_memory_megabytes,
```

Reply:
I am handling this edge case in the code, just fyi.